stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/ui/common/feature/messages/list/DateSeparatorHandler.kt

```diff
@@ -38,60 +40,37 @@ public fun interface DateSeparatorHandler {
 
     public companion object {
 
+        private val defaultDateSeparatorHandler = DateSeparatorHandler { previousMessage, message ->
+            !message.getCreatedAtOrDefault(NEVER).isInTheSameDay(
+                previousMessage?.getCreatedAtOrNull() ?: NEVER,
+            )
+        }
+
         /**
-         * @param separatorTimeMillis Time difference between two message after which we add the date separator.
+         * Creates a [DateSeparatorHandler] returning true if the messages are not in the same day.
          *
          * @return The default normal list date separator handler.
          */
-        public fun getDefaultDateSeparatorHandler(
-            separatorTimeMillis: Long = DateSeparatorDefaultHourThreshold,
-        ): DateSeparatorHandler = DateSeparatorHandler { previousMessage, message ->
-            if (previousMessage == null) {
-                true
-            } else {
-                shouldAddDateSeparator(previousMessage, message, separatorTimeMillis)
-            }
-        }
+        public fun getDefaultDateSeparatorHandler(): DateSeparatorHandler = defaultDateSeparatorHandler
 
         /**
-         * @param separatorTimeMillis Time difference between two message after which we add the date separator.
+         * Creates a [DateSeparatorHandler] returning true if the messages are not in the same day.
          *
          * @return The default thread date separator handler.
          */
-        public fun getDefaultThreadDateSeparatorHandler(
-            separatorTimeMillis: Long = DateSeparatorDefaultHourThreshold,
-        ): DateSeparatorHandler = DateSeparatorHandler { previousMessage, message ->
-            if (previousMessage == null) {
-                false
```

Comment:
Quick question: Previously we didn't show the date separator before the first message in a thread view, but now the separator will always be shown - just checking if that change was intentional?

Reply:
Good Catch!!!
Fixed